code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -200,6 +220,17 @@
 	use_power(power_usage)
 	update_parents()
 
+/obj/machinery/atmospherics/components/unary/thermomachine/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/assembly/signaler/anomaly/pyro))
```

Comment:
This should probably always `return` rather than going through `..()`

Reply:
Fixed and tested.